ksql-engine/src/main/java/io/confluent/ksql/function/udaf/topk/LongTopkKudaf.java

```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright 2017 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package io.confluent.ksql.function.udaf.topk;
+
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaBuilder;
+import org.apache.kafka.streams.kstream.Merger;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import io.confluent.ksql.function.KsqlAggregateFunction;
+import io.confluent.ksql.parser.tree.Expression;
+import io.confluent.ksql.util.ArrayUtil;
+import io.confluent.ksql.util.KsqlException;
+
+
+public class LongTopkKudaf extends KsqlAggregateFunction<Long, Long[]> {
+
+  Integer tkVal;
+  Long[] topkArray;
+  Long[] tempTopkArray;
+
+  public LongTopkKudaf(Integer argIndexInValue, Integer tkVal) {
+    super(argIndexInValue, new Long[tkVal], SchemaBuilder.array(Schema.INT64_SCHEMA).build(),
+          Arrays.asList(Schema.INT64_SCHEMA),
+          "TOPK", LongTopkKudaf.class);
+    this.tkVal = tkVal;
+    this.topkArray = new Long[tkVal];
+    this.tempTopkArray = new Long[tkVal + 1];
+  }
+
+  @Override
+  public Long[] aggregate(Long currentVal, Long[] currentAggVal) {
+    if (currentVal == null) {
```

Comment:
This method and the next look identical to the one in `DoubleTopKudaf` refactor the common parts so the algorithm is only expressed once and can be changed once in the future